src/main/java/com/nha/abdm/wrapper/hip/hrp/GatewayCallbackController.java

```diff
@@ -183,6 +193,37 @@ public ResponseEntity<GatewayCallbackResponse> onAddCareContext(
     return new ResponseEntity<>(HttpStatus.ACCEPTED);
   }
 
+  @PostMapping({"/v0.5/consents/hip/notify"})
+  public ResponseEntity<GatewayCallbackResponse> hipNotify(
+      @RequestBody HIPNotifyRequest hipNotifyRequest) {
+    if (hipNotifyRequest != null
+        && hipNotifyRequest.getHIPNotification() != null
+        && hipNotifyRequest.getHIPNotification().getConsentDetail() != null
+        && hipNotifyRequest.getHIPNotification().getConsentDetail().getPatient() != null) {
+      patientService.addConsentArtefacts(hipNotifyRequest);
+      HIPOnNotifyRequest hipOnNotifyRequest =
+          HIPOnNotifyRequest.builder()
+              .requestId(hipNotifyRequest.getRequestId())
```

Comment:
The `requestId` must be new for every new http request that is issued to Gateway. In this case as per the ABDM documentation, the old requestId can be sent in the field `resp.requestId`, [reference](https://sandbox.abdm.gov.in/abdm-docs/V05_api#post-/v0.5/consents/hip/on-notify).

Reply:
Added a TODO to clarify this.